rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -393,7 +405,7 @@ class ReplayRSpace[F[_], C, P, E, A, R, K](store: IStore[C, P, A, K], branch: Br
     for {
       isEmpty <- syncF.delay(replayData.isEmpty)
       checkpoint <- isEmpty.fold(
-                     Checkpoint(store.createCheckpoint(), Seq.empty).pure[F], {
+                     syncF.delay { Checkpoint(store.createCheckpoint(), Seq.empty) }, {
```

Comment:
Isn't this also pure?

Reply:
I believe `store.createCheckpoint(),` isn't.